Code/Tools/SceneAPI/SceneUI/RowWidgets/NodeTreeSelectionWidget.cpp

```diff
@@ -208,54 +208,57 @@ namespace AZ
+                tempList->EnumerateSelectedNodes(
+                    [&](const AZStd::string& nodeName)
                     {
-                        continue;
-                    }
-
-                    AZStd::shared_ptr<const DataTypes::IGraphObject> object = graph.GetNodeContent(index);
-                    if (!object)
-                    {
-                        continue;
-                    }
+                        Containers::SceneGraph::NodeIndex index = graph.Find(nodeName);
+                        if (!index.IsValid())
+                        {
+                            return true;
+                        }
 
-                    if (m_filterTypes.empty() && m_filterVirtualTypes.empty())
-                    {
-                        result++;
-                        continue;
-                    }
+                        AZStd::shared_ptr<const DataTypes::IGraphObject> object = graph.GetNodeContent(index);
+                        if (!object)
+                        {
+                            return true;
+                        }
 
-                    bool foundType = false;
-                    for (const Uuid& type : m_filterTypes)
-                    {
-                        if (object->RTTI_IsTypeOf(type))
+                        if (m_filterTypes.empty() && m_filterVirtualTypes.empty())
                         {
                             result++;
-                            foundType = true;
-                            break;
+                            return true;
                         }
-                    }
-                    if (foundType)
-                    {
-                        continue;
-                    }
 
-                    // Check if the object is one of the registered virtual types.
-                    AZStd::set<Crc32> virtualTypes;
-                    Events::GraphMetaInfoBus::Broadcast(&Events::GraphMetaInfo::GetVirtualTypes, virtualTypes, *root->GetScene(), index);
-                    for (Crc32 name : virtualTypes)
-                    {
-                        if (m_filterVirtualTypes.find(name) != m_filterVirtualTypes.end())
+                        bool foundType = false;
+                        for (const Uuid& type : m_filterTypes)
                         {
-                            result++;
-                            break;
+                            if (object->RTTI_IsTypeOf(type))
+                            {
+                                result++;
+                                foundType = true;
+                                break;
+                            }
                         }
-                    }
-                }
+                        if (foundType)
+                        {
+                            return true;
+                        }
+
+                        // Check if the object is one of the registered virtual types.
+                        AZStd::set<Crc32> virtualTypes;
```

Comment:
This can be an unordered_set as well
A Crc32 value is not really meaningful in a weakly ordered way, so storing them in a ordered set doesn't provide a benefit
```suggestion
                        AZStd::unordered_set<Crc32> virtualTypes;
```

Reply:
This is coming in another PR, where I've made all the virtual type lists an unordered_set (and gave it an alias), so I won't change it here.

Reply:
Other PR has been submitted